### PR TITLE
Changes for lastest-ci

### PR DESCRIPTION
**Pull request by joshmoore**
These are the "(upstream)" changes from latest-ci making use of https://github.com/ome/jenkins-library-recursivemerge/blob/master/vars/recursiveMerge.groovy. Propose to spin up idr-ci based on this code and once working release as 0.12.1.

### CODE REVIEW

**Comment by joshmoore** [comment]
Manual steps that were/are required:
 - set Oracle password
 - create branch on snoopy/bio-formats-build

File: home/jobs/OMERO-docs/config.xml [hunk 1]
```diff
@@ -31,7 +32,7 @@
     </userRemoteConfigs>
     <branches>
       <hudson.plugins.git.BranchSpec>
-        <name>SPACEBRANCH</name>
+        <name>develop</name>
```
Comment (sbesson) [inline home/jobs/OMERO-docs/config.xml]:
```suggestion
        <name>SPACEBRANCH</name>
```

File: home/jobs/OMERO-push/config.xml [hunk 2]
```diff
@@ -41,7 +41,7 @@
     </userRemoteConfigs>
     <branches>
       <hudson.plugins.git.BranchSpec>
-        <name>SPACEBRANCH</name>
+        <name>develop</name>
```
Comment (sbesson) [inline home/jobs/OMERO-push/config.xml]:
```suggestion
        <name>SPACEBRANCH</name>
```

**Comment by joshmoore** [comment]
> A few inline suggestions.

Sorry, I should have raised this for discussion. Since we aren't currently using an integration branch strategy, I propose to hard code these branches to remove one of the manual steps that I stumble over each time I spin up a devspace.

**Comment by sbesson** [comment]
Ah understood. It makes sense. I am also in favor of limiting the number of manual steps/customizations while deploying devspaces. If pointing to a non-default is needed, I think the strategy used in BIOFORMATS-push can be applied:

https://github.com/ome/devspace/blob/c381283d1461da7b152ac6a7b56f82f90d7ddf29/home/jobs/BIOFORMATS-push/config.xml#L32-L38

Should we update the `merge SPACEBRANCH --no-ask --reset` parameters in these jobs instead? Once this PR is merged, we can review https://github.com/ome/devspace/search?q=SPACEBRANCH&unscoped_q=SPACEBRANCH.

**Comment by sbesson** [comment]
Just realized the current state is breaking the Bio-Formats -> OMERO coupling. This is primarily because of the nature of `BIOFORMATS-build` as a matrix job which breaks expectations with regard to `copyArtifacts`. Merging and I will look into migrating the version coupling between the push jobs in a follow-up PR.

**Pull request merged**